src/points/slice.ts

```diff
@@ -92,12 +98,14 @@ const slice = createSlice({
       ...state,
       pendingPointsEvents: [...state.pendingPointsEvents, action.payload],
     }),
-    pointsEventProcessed: (state, action: PayloadAction<Pick<PendingPointsEvent, 'id'>>) => ({
-      ...state,
-      pendingPointsEvents: state.pendingPointsEvents.filter(
+    pointsEventProcessed: (state, action: PayloadAction<PendingPointsEvent>) => {
+      state.pendingPointsEvents = state.pendingPointsEvents.filter(
         (event) => event.id !== action.payload.id
-      ),
-    }),
+      )
+      if (action.payload.event.activityId in state.trackOnceActivities) {
+        state.trackOnceActivities[action.payload.event.activityId] = true
+      }
+    },
```

Comment:
Shouldn't we add it to `trackOnceActivities` when `trackPointsEvent` is dispatched?
It's guaranteed it will be processed because there's a queue, right?

Otherwise it's possible we dispatch more than once in a session. Though unlikely if requests are fast.

Reply:
yeah you're right, i can move this further up. i think it can't be as far up as `trackPointsEvent` though because we need a chance to check whether the activity has been tracked. i've moved it to be updated on `sendPointsEventStarted`